nodejs/simple-cli/upload.js

```diff
@@ -0,0 +1,169 @@
+/// This example project aims to demonstrate how to integrate the `w3up-client` SDK
+/// into a node command line app. 
+///
+/// To keep the example focused, we'll be leaving a lot of things out that you'd
+/// probably want to have in a "production-ready" application. For a real-world
+/// CLI example, see https://github.com/web3-storage/w3up-cli
+
+/// First we need to import the `createClient` function from the `w3up-client` package.
+import { createClient } from 'w3up-client'
+
+/// We also import the `packToBob` function from the `ipfs-car` package, 
+/// to create IPFS-formatted file objects and pack them into a Content Archive (CAR) for uploading. 
+/// Note that this will likely be handled automatically by a future version of `w3up-client`.
+import { packToBlob } from 'ipfs-car/pack/blob'
+
+/// Finally, we import a few things from the node standard library
+import fs from 'fs'
+import path from 'path'
+
+/// In this example, we always load the identity key from a `settings.json` file
+/// in the current directory. See the `loadSettings` function below.
+const SETTINGS_FILENAME = 'settings.json'
+
+/**
+ * The entry-point function is invoked at the bottom of this file.
+ */
+async function main() {
+  // To keep things super simple, we only support a single command line argument: the path to a file.
+  // We need to slice off the first two elements of process.args, which contain the nodejs command and path to the script file.
+  const args = process.argv.slice(2)
+
+  // Bail out if we have an unexpected number of arguments
+  if (args.length !== 1) {
+    const scriptPath = path.relative(process.cwd(), process.argv[1])
+    console.error(`usage: node ${scriptPath} <filename>`)
+    process.exit(1)
+  }
+
+  const filename = args[0]
+
+  // Load the secret key from a local settings.json file.
+  // If it does not exist, `loadIdentity` will prompt the user to
+  // register an identity using the `register.js` script.
+  const { secret, email } = await loadIdentity()
+
+  // create a new w3up client using the secret key
+  const client = makeClientWithRegisteredSecretKey(secret)
+
+  // client.identity() returns an object we can use to print the `did:key:`
+  // string that is used as our public identity.
+  const id = await client.identity()
+  console.log(`Uploading as ${id.did()} (${email})`)
+
+  const { rootCID, carBytes } = await prepareForUpload(filename)
+  console.log(`Uploading ${filename}`)
+  console.log(`Root CID: ${rootCID}\n`)
+
+  try {
+    await client.upload(carBytes)
+    console.log('🎉 Upload complete!')
```

Comment:
Should this completion stuff come after the try block? 
With no internet I get an error AND the upload complete message because of this.
```
Uploading as did:key:z6MkopzHycxGVBSPsGCHozUHJcbKx1AjDudv6w1cPtDtp5QM (josh.jarvis+10@potatolondon.com)
Uploading 10MB.zip
Root CID: bafybeie4kwih4tm726ikfutek657bicyjhplcdi2w7hc36dbengbtc4o2e

FetchError: request to https://8609r1772a.execute-api.us-east-1.amazonaws.com/ failed, reason: getaddrinfo ENOTFOUND 8609r1772a.execute-api.us-east-1.amazonaws.com
    at ClientRequest.<anonymous> (/Users/josh/sites/potato/protocal-labs/w3up-client-examples/nodejs/simple-cli/node_modules/node-fetch/lib/index.js:1491:11)
    at ClientRequest.emit (node:events:527:28)
    at TLSSocket.socketErrorListener (node:_http_client:454:9)
    at TLSSocket.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  type: 'system',
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND'
}
🎉 Upload complete!
🔗 View on an IPFS <-> HTTP gateway at https://w3s.link/ipfs/bafybeie4kwih4tm726ikfutek657bicyjhplcdi2w7hc36dbengbtc4o2e
Note that it may take a few seconds before content is available.
```

Reply:
I think was caused by the client code - it was logging but not re-throwing unexpected errors. It [seems to have been updated tho](https://github.com/web3-storage/w3up-client/blob/main/src/index.js#L289), so after updating it seems to Fail Successfully :)

```
Uploading as did:key:z6MknvLxgd6Gqnf7S5aYnGfx6oYUxuVJMMmXhbXWtGcHJKep (yusef+test@dag.house)
Uploading README.md
Root CID: bafybeibtvxeihc4plutzd5tessno4uza73fna6yejshe3iwxzmmrlg37da

FetchError: request to https://8609r1772a.execute-api.us-east-1.amazonaws.com/ failed, reason: getaddrinfo ENOTFOUND 8609r1772a.execute-api.us-east-1.amazonaws.com
    at ClientRequest.<anonymous> (/home/yusef/work/repos/w3up-client-examples/nodejs/simple-cli/node_modules/node-fetch/lib/index.js:1461:11)
    at ClientRequest.emit (node:events:527:28)
    at TLSSocket.socketErrorListener (node:_http_client:454:9)
    at TLSSocket.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  type: 'system',
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND'
}
🧨 Upload failed with an error: FetchError: request to https://8609r1772a.execute-api.us-east-1.amazonaws.com/ failed, reason: getaddrinfo ENOTFOUND 8609r1772a.execute-api.us-east-1.amazonaws.com
```